File: Frontend/Lafatkotob-SPA/src/app/Auth/Models/AppUserModel.ts
export interface AppUserModel {
  id: string;
  isDeleted: boolean;
  name: string;
  city: string;
  email: string;
  dateJoined: Date;
  lastLogin: Date;
  profilePicture: string;
  about: string;
  dthDate: Date;
  historyId?: number;
  upVotes?: number;
}

File: Backend/Lafatkotob.API/Lafatkotob/ViewModels/AppUserModel.cs
using Lafatkotob.Entities;

namespace Lafatkotob.ViewModels
{
    public class AppUserModel
    {
        public string Id { get; set; }
        public bool IsDeleted { get; set; }
        public string Name { get; set; }
        public string City { get; set; }
        public string Email { get; set; }
        public DateTime DateJoined { get; set; }
        public DateTime LastLogin { get; set; }
        public string ProfilePicture { get; set; }
        public string About { get; set; }
        public DateTime DTHDate { get; set; }
        public int? HistoryId { get; set; }
        public int? UpVotes { get; set; }

        public int Age => DateTime.Today.Year - DTHDate.Year - (DateTime.Today.DayOfYear < DTHDate.DayOfYear ? 1 : 0);

 
    }
}
